tooling/minimize/tests/ub/tree_borrows/protector/reservedim_spurious_write.rs

```diff
@@ -0,0 +1,84 @@
+//@ compile-flags: --minimize-tree-borrows
+//@revisions: without with
+
+// This test was taken from Miri Tree Borrows
+// https://github.com/rust-lang/miri/blob/master/tests/fail/tree_borrows/reservedim_spurious_write.rs
+
+// Illustrating a problematic interaction between Reserved, interior mutability, and protectors.
+
+// One revision without spurious read (default source code) and one with spurious read.
+// Both are expected to be UB. Both revisions are expected to have the *same* error
+// because we are aligning the behavior of `without` to that of `with` so that the
+// spurious write is effectively a noop in the long term.
+
+extern crate intrinsics;
+use intrinsics::*;
+use std::cell::Cell;
+
+#[derive(Clone, Copy)]
+struct ThreadData {
+    counter_ptr: *mut u32,
+    data_ptr: *mut u8,
+}
+
+// Create a lazy Reserved with interior mutability.
+// Wait for the other thread's spurious write then observe the side effects
+// of that write.
+extern "C" fn thread(thread_data_ptr: *const ()) {
+    let ThreadData { counter_ptr, data_ptr }  = unsafe { *(thread_data_ptr as *const ThreadData) };
+
+    fn inner(y: &mut Cell<()>, counter_ptr: *mut u32) -> *mut u8 {
+        unsafe { atomic_fetch_add(counter_ptr, 1); } // 1 -> 2
```

Comment:
```suggestion
        assert!(unsafe { atomic_fetch_add(counter_ptr, 1) } == 1); // 1 -> 2
```
This makes sure that the comment is actually correct.